File: zd1.swift
import Foundation

class StringValidator {
    static func main() {
        print("Введите текст")
        if let text = readLine() {
            var modifiedText = text
            for i in 0..<(text.count - 2) {
                let index1 = text.index(text.startIndex, offsetBy: i)
                let index2 = text.index(text.startIndex, offsetBy: i + 1)
                let index3 = text.index(text.startIndex, offsetBy: i + 2)
                
                if text[index1] == text[index2] && text[index2] == text[index3] {
                    modifiedText.remove(at: index3)
                    modifiedText.insert(randomChar(), at: index3)
                }
            }
            print(modifiedText)
        }
    }
    
    static func randomChar() -> Character {
        let chars = Array("abcdefghijklmnopqrstuvwxyz")
        return chars[Int.random(in: 0..<chars.count)]
    }
}

StringValidator.main()

File: zd1.rb
class StringValidator
  def self.main
    puts "Введите текст"
    text = gets.chomp
    if text.nil?
      puts "у вас пустой текст"
    else
      (0..text.length - 3).each do |i|
        if text[i] == text[i + 1] && text[i + 1] == text[i + 2]
          text[i + 2] = random_char
        end
      end
    end
    puts text
  end

  def self.random_char
    chars = ("a".."z").to_a
    chars.sample
  end
end

StringValidator.main
